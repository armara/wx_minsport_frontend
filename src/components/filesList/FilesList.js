import React from 'react'
import PropTypes from 'prop-types'

import FileRow from 'components/fileRow/FileRow'
import List from 'components/list/List'
import Button from 'components/button/Button'
import FileBase64 from 'components/fileBase64/FileBase64'
import ToolTipIcon from 'components/toolTip/icon/ToolTipIcon'

import './FilesList.scss'

const FilesList = ({
  items,
  headerText,
  shouldAddMore,
  className,
  onRowClick,
  onAddClick,
  accept,
  fileTypeDataTip,
  addMoreDataTip,
}) => (
  <div className="files-list">
    <h4>
      {headerText}{' '}
      {fileTypeDataTip && <ToolTipIcon dataTip={fileTypeDataTip} />}{' '}
    </h4>
    {items.length > 0 && (
      <List items={items} className={className}>
        {({ id, name, actionName }) => (
          <FileRow
            fileName={name}
            buttonText={actionName || 'Удалить'}
            onClick={() => onRowClick(id)}
          />
        )}
      </List>
    )}
    {shouldAddMore && (
      <>
        <Button
          className="inline-button"
          onClick={() =>
            document.querySelectorAll('input[type="file"]')[0].click()
          }
        >
          Добавить еще файл
        </Button>
        {addMoreDataTip && <ToolTipIcon dataTip={addMoreDataTip} />}
        <FileBase64 onDone={onAddClick} accept={accept} />
      </>
    )}
  </div>
)

FilesList.defaultProps = {
  className: '',
  headerText: 'Файлы',
  shouldAddMore: false,
  onRowClick: () => {},
  onAddClick: () => {},
  accept: null,
  fileTypeDataTip: '',
  addMoreDataTip: '',
}

FilesList.propTypes = {
  className: PropTypes.string,
  shouldAddMore: PropTypes.bool,
  items: PropTypes.arrayOf(PropTypes.object).isRequired,
  headerText: PropTypes.string,
  onRowClick: PropTypes.func,
  onAddClick: PropTypes.func,
  accept: PropTypes.string,
  fileTypeDataTip: PropTypes.string,
  addMoreDataTip: PropTypes.string,
}

export default FilesList
