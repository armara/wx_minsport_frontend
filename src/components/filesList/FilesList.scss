@import 'styles/mixins';

.files-list {
  h4 {
    @include nav-header-text($x-small, '18px', $evil-gray);
    margin-top: 10px;
    margin-bottom: 10px;

    .tooltip-icon {
      display: inline-flex;
      position: relative;
      top: 5px;
    }
  }

  button + div {
    display: inline-flex;
    margin-top: 8px;
  }

  .contract-files {
    @include contracts-block();
  }

  input[type='file'] {
    display: none;
  }
}
