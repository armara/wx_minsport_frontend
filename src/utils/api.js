import axios from 'axios'
import config from 'config'

import { getUserWorkplaceId } from 'utils/user'

const facilityId = getUserWorkplaceId()
const {
  BACKEND_URL,
  auth: { username, password },
  service: {
    auth,
    sports,
    districts,
    facilitiesV2,
    federations,
    organizations,
    personsV3,
    scheduleV2,
  },
} = config

const authUrl = BACKEND_URL + auth
const facilitiesv2Url = BACKEND_URL + facilitiesV2
const districtsUrl = BACKEND_URL + districts
const sportsUrl = BACKEND_URL + sports
const scheduleV2Url = BACKEND_URL + scheduleV2
const federationUrl = BACKEND_URL + federations
const personsV3Url = BACKEND_URL + personsV3
const organizationsUrl = BACKEND_URL + organizations

const axiosAuth = axios.create({
  baseURL: authUrl,
  auth: {
    username,
    password,
  },
})

export default () => ({
  auth: formData => axiosAuth.post(`/token`, formData),

  getOrganizationById: id =>
    axios.get(`${organizationsUrl}/organizations/${id}`),
  getFederationById: id => axios.get(`${federationUrl}/federation/${id}`),

  postGroup: body => axios.post(`${personsV3Url}/group`, { ...body }),
  putGroup: (body, id) => axios.put(`${personsV3Url}/group/${id}`, { ...body }),
  deleteGroup: (id, body) =>
    axios.delete(`${personsV3Url}/group/${id}`, {
      data: body,
      headers: {
        'Content-Type': 'application/json',
      },
    }),

  getDistricts: () => axios.get(`${districtsUrl}/district/all`),
  getSubdistricts: districtId =>
    axios.get(`${districtsUrl}/subdistrict/allByDistrict/`, {
      params: {
        districtId,
      },
    }),

  getGroups: query => axios.get(`${personsV3Url}/group${query}`),
  getPersons: query => axios.get(`${personsV3Url}/person${query}`),
  updatePerson: (personId, body) =>
    axios.put(`${personsV3Url}/person/${personId}`, { ...body }),
  getGroupById: id => axios.get(`${personsV3Url}/group/${id}`),
  getPersonById: id => axios.get(`${personsV3Url}/person/${id}`),
  getPositions: () => axios.get(`${personsV3Url}/position`),
  addPerson: body => axios.post(`${personsV3Url}/person`, { ...body }),
  removePerson: (id, body) =>
    axios.delete(`${personsV3Url}/person/${id}`, {
      data: body,
      headers: {
        'Content-Type': 'application/json',
      },
    }),
  getFederations: () => axios.get(`${federationUrl}/federations`),
  getPersonAvatarUrl: personId => `${personsV3Url}/person/${personId}/avatar`,

  uploadPersonAvatar2: (personId, formData) =>
    axios.post(`${personsV3Url}/person/${personId}/avatar`, formData, {
      headers: {
        'Content-Type': 'multipart/form-data',
      },
    }),

  getRanks: () => axios.get(`${personsV3Url}/rank`),
  getGroupsByPersonId: (query = '') =>
    axios.get(`${personsV3Url}/group/${query}`),

  getEventTypes: () => axios.get(`${scheduleV2Url}/types/events`),
  addEventItem: body => axios.post(`${scheduleV2Url}/events`, { ...body }),

  getScheduleEventById: id => axios.get(`${scheduleV2Url}/events/${id}`),
  getScheduleById: id => axios.get(`${scheduleV2Url}/schedules/${id}`),
  editScheduleItem: (body, id) =>
    axios.put(`${scheduleV2Url}/schedules/${id}`, { ...body }),

  getFacilitySports: _facilityId =>
    axios.get(`${sportsUrl}/fsports/`, {
      params: {
        facilityId: _facilityId || facilityId,
        pageSize: 10000,
        returnDisciplines: true,
      },
    }),
  getFacilityDisciplines: () =>
    axios.get(
      `${sportsUrl}/fdisciplines/
      `,
      {
        params: {
          facilityId,
          pageSize: 10000,
        },
      }
    ),
  getSportsBulk: ids => axios.post(`${sportsUrl}/sports/bulk`, { ids }),
  getSportBy: id => axios.get(`${sportsUrl}/sports/${id}`),
  getFacilitySportBy: id => axios.get(`${sportsUrl}/fsports/${id}`),
  getFullSports: (params = {}) =>
    axios.get(`${sportsUrl}/sports/`, {
      params,
    }),
  getFullDisciplines: () => axios.get(`${sportsUrl}/disciplines`),

  getAllFacilities: () => axios.get(`${facilitiesv2Url}/facility/all`),
  getAreas: () => axios.get(`${facilitiesv2Url}/facility/${facilityId}/areas`),
  getAbonnements: abonnementId =>
    axios.get(`${facilitiesv2Url}/facility/abonnement/${abonnementId}`),
  getAllAbonnements: () =>
    axios.get(`${facilitiesv2Url}/facility/${facilityId}/abonnements`),
  getZones: areaId =>
    axios.get(`${facilitiesv2Url}/facility/area/${areaId}/zones`),
  getFacilityByIdSummary: () =>
    axios.get(`${facilitiesv2Url}/facility/${facilityId}/summary`),
  getFacilityById: id => axios.get(`${facilitiesv2Url}/facility/${id}`),
  getOrgTypes: () => axios.get(`${organizationsUrl}/organizations/types`),
  getAllOrganizations: () => axios.get(`${organizationsUrl}/organizations`),
  postOrganization: body =>
    axios.post(`${organizationsUrl}/organizations`, body),
  getOrganizationByQuery: params =>
    axios.get(`${organizationsUrl}/organizations/`, { params }),
  getContractTypes: () =>
    axios.get(`${facilitiesv2Url}/facility/contract/type/all`),
  getContractFiles: contractId =>
    axios.get(`${facilitiesv2Url}/facility/contract/${contractId}/files`),
  getZonesByContractId: contractId =>
    axios.get(`${facilitiesv2Url}/facility/contract/${contractId}/zones`),
  postContract: body =>
    axios.post(`${facilitiesv2Url}/facility/contract`, body),
  putContract: contract =>
    axios.put(`${facilitiesv2Url}/facility/contract`, contract),
  postZone: zone =>
    axios.post(`${facilitiesv2Url}/facility/contract/zone`, zone),
  removeZones: (contractId, reason = '') =>
    axios.delete(`${facilitiesv2Url}/facility/contract/${contractId}/zones`, {
      data: { reason },
    }),
  getAllContracts: () =>
    axios.get(`${facilitiesv2Url}/facility/${facilityId}/contracts`),
  getAllOrganizationContracts: orgId =>
    axios.get(`${facilitiesv2Url}/facility/contracts?orgId=${orgId}`),
  getContract: id => axios.get(`${facilitiesv2Url}/facility/contract/${id}`),
  removeContract: (contractId, reason = '') =>
    axios.delete(`${facilitiesv2Url}/facility/contract/${contractId}`, {
      data: { reason },
    }),
  postFile: file =>
    axios.post(`${facilitiesv2Url}/facility/contract/file`, file),
  removeFile: (id, reason = '') =>
    axios.delete(`${facilitiesv2Url}/facility/contract/file/${id}`, {
      data: { reason },
    }),

  getEvents: query => axios.get(`${scheduleV2Url}/events${query}`),
  getSchedules: query =>
    axios.get(`${scheduleV2Url}/schedules${query}`, {
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Content-Type': 'application/json',
      },
    }),

  removeSchedule: (id, body) =>
    axios.delete(`${scheduleV2Url}/schedules/${id}`, {
      data: body,
    }),

  removeEvent: (id, body) =>
    axios.delete(`${scheduleV2Url}/events/${id}`, {
      data: body,
    }),
})

async function requestIfIdDefined(request, id) {
  let response = null
  try {
    if (id) {
      const { data } = await request(id)
      response = data
    }
    /* eslint-disable-next-line */
  } catch {}
  return response
}

export { requestIfIdDefined }
