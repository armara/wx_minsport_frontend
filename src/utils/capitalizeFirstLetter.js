export default string => string.replace(string[0], string[0].toUpperCase())
