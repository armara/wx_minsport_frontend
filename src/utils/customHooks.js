import { useEffect } from 'react'

const useInitData = (data, handleFetch) => {
  useEffect(() => {
    if (data.length === 0) {
      handleFetch()
    }
  }, [data, handleFetch])
}

const useInitNoCondition = (data, handleEffect) => {
  useEffect(() => {
    handleEffect(data)
  }, [data, handleEffect])
}

const useInitByCondition = (data, handleEffect, customCondition) => {
  useEffect(() => {
    if (customCondition) {
      handleEffect(data)
    }
  }, [data, handleEffect, customCondition])
}

export { useInitData as default, useInitNoCondition, useInitByCondition }
