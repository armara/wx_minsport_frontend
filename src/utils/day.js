import 'dayjs/locale/ru'
import customParseFormat from 'dayjs/plugin/customParseFormat'

import dayjs from 'dayjs'
import { DAY_FORMAT } from './localization'

dayjs.extend(customParseFormat)
dayjs.locale('ru')

export const currentDay = dayjs().format(DAY_FORMAT)

export const dayDiff = (date1, date2) => date1.diff(date2, 'day')
export const timeDiff = (date1, date2) => date1.diff(date2, 'minutes')
export const secondsDiff = (date1, date2) => date1.diff(date2, 'second')
export const ageFrom = millis => dayjs().diff(dayjs(millis), 'year')

export const hasTimePassed = (startTime, period) => {
  const day = dayjs()
  const startDay = dayjs(startTime)
  const diff = secondsDiff(day, startDay)
  return diff > period
}

export const getDateValueOrNull = date => {
  const dayjsDate = dayjs(date, DAY_FORMAT)
  if (!dayjsDate.isValid()) {
    return null
  }

  return dayjsDate.valueOf()
}

export default dayjs
