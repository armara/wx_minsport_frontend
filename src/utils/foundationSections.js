import editorImage from 'assets/images/placeholder.jpg'

export const FoundationItems = ({
  legalData,
  orgType,
  organizationName,
  tin,
  description,
  contractTitle,
  sportTitle,
  reg,
  end,
}) => [
  {
    field: 'Название контракта',
    value: legalData,
    key: 'contract-legal-data',
  },
  {
    field: 'Описание контракта',
    value: description,
    key: 'contract-description',
  },
  {
    field: 'Название организации',
    value: organizationName,
    key: 'organization-name',
  },
  {
    field: 'Тип организации',
    value: (orgType || {}).name,
    key: 'foundation-type',
  },
  {
    field: 'ИНН',
    value: tin,
    key: 'organization-tin',
  },
  {
    field: 'Вид спорта',
    value: sportTitle,
    key: 'sport-title',
  },
  {
    field: 'Классификатор договора',
    value: contractTitle,
    key: 'contract-classifier',
  },
  {
    field: 'Дата заключения договора',
    value: reg,
    key: 'contract-start-date',
  },
  {
    field: 'Дата окончания договора',
    value: end,
    key: 'contract-expire-date',
  },
  {
    type: 'hr',
    key: 'hr-2',
  },
]

export const FoundationTeamSections = [
  {
    title: 'Спортсмены',
    key: 'sportsmen',
  },
  {
    title: 'Тренеры',
    key: 'coaches',
  },
]

export const FoundationSportsmen = [
  {
    image: editorImage,
    field: 'Ковалев В.А.',
    value: 'Мастер спорта',
    key: 'sportsmen-1',
  },
  {
    image: editorImage,
    field: 'Ромодановский Ф.',
    value: 'Заслуженный мастер спорта',
    key: 'sportsmen-2',
  },
  {
    image: editorImage,
    field: 'Франц Лефорт',
    value: 'Мастер спорта',
    key: 'sportsmen-3',
  },
  {
    image: editorImage,
    field: 'Нарышкин П.А.',
    value: 'Мастер спорта международного класса',
    key: 'sportsmen-4',
  },
  {
    image: editorImage,
    field: 'Шереметев Б.П.',
    value: 'Мастер спорта международного класса',
    key: 'sportsmen-5',
  },
  {
    image: editorImage,
    field: 'Штейн К.',
    value: 'Заслуженный мастер спорта',
    key: 'sportsmen-6',
  },
  {
    image: editorImage,
    field: 'Гусин Л.Д.',
    value: 'Мастер спорта',
    key: 'sportsmen-7',
  },
  {
    image: editorImage,
    field: 'Плещеев И.А.',
    value: 'Мастер спорта международного класса',
    key: 'sportsmen-8',
  },
]

export const FoundationFiles = [
  {
    name: 'Договор',
    key: 'contract',
  },
  {
    name: 'Приложение к договору',
    key: 'contract-addon',
  },
]
