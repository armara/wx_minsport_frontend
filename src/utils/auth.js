import axios from 'axios'
import debounce from 'lodash.debounce'

import { getUserData, createWorkplace, updateUserData } from 'utils/user'
import api from 'utils/api'
import history from 'utils/utils'
import dayjs, { hasTimePassed } from 'utils/day'

const MILLISECONDS_BEFORE_REFRESH_REQUEST = 1000

export const isAuthenticated = () => Object.keys(getUserData()).length > 0

const getUserDataFromResponse = response => {
  const {
    data: {
      access_token: accessToken,
      email,
      jti,
      phone,
      facilities,
      federations,
      organizations,
      ministries,
      first_name: firstName,
      sur_name: surName,
      patronymic_name: patronymicName,
      expires_in: expiresIn,
      refresh_token: refreshToken,
      token_type: tokenType,
    } = {},
  } = response

  const facilitiesIds = Object.keys(facilities)
  const federationsIds = Object.keys(federations)
  const organizationsIds = Object.keys(organizations)
  const ministriesIds = Object.keys(ministries)

  return {
    accessToken,
    email,
    jti,
    firstName: firstName || '',
    surName: surName || '',
    patronymicName: patronymicName || '',
    phone,
    facilities: facilitiesIds,
    federations: federationsIds,
    organizations: organizationsIds,
    ministries: ministriesIds,
    expiresIn,
    refreshToken,
    tokenType,
  }
}

const chooseInitialUserWorkplace = ({
  facilities,
  organizations,
  federations,
  ministries,
}) => {
  let type
  let id
  if (facilities.length > 0) {
    type = 'facilities'
    ;[id] = facilities
  } else if (organizations.length > 0) {
    type = 'organizations'
    ;[id] = organizations
  } else if (federations.length > 0) {
    type = 'federations'
    ;[id] = federations
  } else if (ministries.length > 0) {
    type = 'ministries'
    ;[id] = ministries
  }

  return createWorkplace(type, id)
}

export const login = async ({ username, password }) => {
  const formData = new FormData()
  formData.append('grant_type', 'password')
  formData.append('username', username) // petr007
  formData.append('password', password) // qwerty

  return (
    api()
      .auth(formData)
      .then(response => {
        if (response.status === 200) {
          const userData = getUserDataFromResponse(response)
          const workPlace = chooseInitialUserWorkplace(userData)
          const time = dayjs().toISOString()
          updateUserData(userData, { workPlace, time })
        }

        return response
      })
      // eslint-disable-next-line no-console
      .catch(console.error)
  )
}

export const logout = () => {
  localStorage.removeItem('user')
}

const refreshAccessToken = refreshToken => {
  const formData = new FormData()
  formData.append('grant_type', 'refresh_token')
  formData.append('refresh_token', refreshToken)

  return api()
    .auth(formData)
    .then(response => {
      if (response.status !== 200) {
        logout()
        window.location.assign('/login')
      }

      const currentUserData = getUserData()
      const nextUserData = getUserDataFromResponse(response)
      const time = dayjs().toISOString()
      updateUserData(currentUserData, { ...nextUserData, time })

      return response
    })
    .catch(() => {
      logout()
      window.location.assign('/login')
    })
}

const debouncedRefreshAccessToken = debounce(
  refreshAccessToken,
  MILLISECONDS_BEFORE_REFRESH_REQUEST,
  {
    leading: true,
    trailing: false,
  }
)

const getValidUserToken = async () => {
  const { time, expiresIn, refreshToken } = getUserData()

  const isPassed = hasTimePassed(time, expiresIn)
  if (isPassed) {
    await debouncedRefreshAccessToken(refreshToken)
    debouncedRefreshAccessToken.cancel()
  }

  return getUserData()
}

export const setupInterceptor = () => {
  axios.interceptors.response.use(
    response => {
      return response
    },
    error => {
      return Promise.reject(error)
    }
  )

  axios.interceptors.request.use(
    async config => {
      const isAuthed = isAuthenticated()

      if (config.url.endsWith('/login')) {
        return config
      }

      if (!isAuthed) {
        history.push('/login')
        return config
      }

      const { accessToken, tokenType } = await getValidUserToken()
      return {
        ...config,
        headers: {
          Authorization: `${tokenType} ${accessToken}`,
        },
      }
    },
    error => {
      return Promise.reject(error)
    }
  )
}

export const facilitiesPrefix = 'facilities'
export const organizationsPrefix = 'organizations'
