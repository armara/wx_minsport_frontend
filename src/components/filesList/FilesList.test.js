import React from 'react'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

import FilesList from 'components/filesList/FilesList'
import { FoundationFiles } from 'utils/foundationSections'

configure({ adapter: new Adapter() })

describe('<FilesList />', () => {
  it('renders correctly default props', () => {
    const tree = renderer.create(<FilesList items={FoundationFiles} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
