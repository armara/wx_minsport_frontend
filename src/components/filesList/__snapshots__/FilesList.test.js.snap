// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilesList /> renders correctly default props 1`] = `
<div
  className="files-list"
>
  <h4>
    Файлы
  </h4>
  <ul
    className="base-list"
    id={null}
  >
    <li
      className=""
    >
      <span
        className="file-row_file-block"
      >
        <i
          className="ic-pdf"
        />
        <span>
          Договор
        </span>
      </span>
      <button
        className="file-row_button button base-button"
        disabled={false}
        id={null}
        onClick={[Function]}
        title=""
        type="button"
      >
        
        Удалить
      </button>
    </li>
    <li
      className=""
    >
      <span
        className="file-row_file-block"
      >
        <i
          className="ic-pdf"
        />
        <span>
          Приложение к договору
        </span>
      </span>
      <button
        className="file-row_button button base-button"
        disabled={false}
        id={null}
        onClick={[Function]}
        title=""
        type="button"
      >
        
        Удалить
      </button>
    </li>
  </ul>
</div>
`;
